src/metabase/query_processor/pivot.clj

```diff
@@ -387,6 +410,22 @@
     (when (some some? (vals pivot-opts))
       pivot-opts)))
 
+(mu/defn- pivot-options :- ::pivot-opts
+  "Looks at the `pivot_table.column_split` key in the card's visualization settings and generates `pivot-rows` and
+  `pivot-cols` to use for generating subqueries. Supports both column name and field ref-based settings.
+
+  Field ref-based visualization settings are considered legacy and are not used for new questions. To not break existing
+  questions we need to support both old- and new-style settings until they are fully migrated."
+  [query        :- [:map
+                    [:database ::lib.schema.id/database]]
+   viz-settings :- [:maybe :map]]
+  (let [column-split         (:pivot_table.column_split viz-settings)
+        column-split-rows    (seq (:rows column-split))
+        column-split-columns (seq (:columns column-split))]
+    (if (and (every? string? column-split-rows) (every? string? column-split-columns))
```

Comment:
```suggestion
  (let [{:keys [rows columns]} (:pivot_table.column_split viz-settings)]
    (if (and (every? string? rows) (every? string? columns))
```
You shouldn't need to explicitly call `seq` on the pieces (even if they're maybe `nil`), the implementation of `every?` will do it if needed.